dts/bindings/usb/zephyr,hid-device.yaml

```diff
@@ -0,0 +1,50 @@
+# Copyright (c) 2023 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+description: Bindings for virtual HID device
+
+compatible: "zephyr,hid-device"
+
+include: base.yaml
+
+properties:
+  interface-name:
+    type: string
+    description: |
+      HID device name. When this property is present, a USB device will use it
+      as the string descriptor of the interface.
+
+  boot-protocol-code:
+    type: string
+    required: true
+    description: |
+      Device can be used as a Boot Device and supports a predefined protocol
+      for mouse or keyboards.
+      - none:     Device does not support the boot interface
+      - keyboard: Device supports boot interface and keyboard protocol
+      - mouse:    Device supports boot interface and mouse protocol
+    enum:
+      - none
+      - keyboard
+      - mouse
+
+  polling-rate:
+    type: int
+    required: true
+    description: |
+      Input type reports polling rate in number of 125µs periods.
+      Depending on the actual speed, the calculated polling rate could be clamped
+      to 1ms or 255ms.
+
+  in-reports-mps:
```

Comment:
I think we should have separate max packet size entry for full-speed and high-speed, because the two have different allowed ranges. While the USB stack does not really support the concept of different configurations for different speed we really should have it because otherwise we'll run into unsolvable issues later on.